src/main/java/com/couchbase/intellij/tree/cblite/CBLTreeRightClickListener.java

```diff
@@ -89,58 +100,114 @@ public void actionPerformed(@NotNull AnActionEvent e) {
+        }
+
+        AnAction deleteConnection = new AnAction("Delete Connection") {
             @Override
             public void actionPerformed(@NotNull AnActionEvent e) {
-                int result = Messages.showYesNoDialog("<html>Are you sure you want to delete the document <strong>" + userObject.getId() + "</strong>?</html>", "Delete Document", Messages.getQuestionIcon());
+                try {
+                    CBLTreeHandler.deleteConnection(clickedNode, userObject, tree);
+                } catch (CouchbaseLiteException ex) {
+                    Log.error(ex);
+                    SwingUtilities.invokeLater(() -> Messages.showErrorDialog("Could not delete the connection.",
+                            "Couchbase Lite Plugin Error"));
+                }
+            }
+        };
+        actionGroup.add(deleteConnection);
+        showPopup(e, tree, actionGroup);
+    }
+
+    private static void handleScope(Project project, MouseEvent e, DefaultMutableTreeNode clickedNode,
+            CBLScopeNodeDescriptor userObject, Tree tree) {
+        DefaultActionGroup actionGroup = new DefaultActionGroup();
+
+        AnAction deleteScope = new AnAction("Delete Scope") {
+            @Override
+            public void actionPerformed(@NotNull AnActionEvent e) {
+                int result = Messages.showYesNoDialog("<html>Are you sure you want to delete the scope <strong>"
+                        + userObject.getText() + "</strong>?</html>", "Delete Scope", Messages.getQuestionIcon());
                 if (result != Messages.YES) {
                     return;
                 }
 
                 try {
-                    Document document = ActiveCBLDatabase.getInstance().getDatabase().
-                            getScope(userObject.getScope())
-                            .getCollection(userObject.getCollection()).getDocument(userObject.getId());
-
-                    ActiveCBLDatabase.getInstance().getDatabase().
-                            getScope(userObject.getScope())
-                            .getCollection(userObject.getCollection()).delete(document);
-
-                    if (userObject.getVirtualFile() != null) {
-                        try {
-                            FileEditorManager fileEditorManager = FileEditorManager.getInstance(project);
-                            fileEditorManager.closeFile(userObject.getVirtualFile());
-                        } catch (Exception ex) {
-                            ex.printStackTrace();
-                            Log.debug("Could not close the file", ex);
-                        }
-                    }
-
-                    DefaultMutableTreeNode parentNode = (DefaultMutableTreeNode) clickedNode.getParent();
-                    if (parentNode != null) {
-                        ((DefaultTreeModel) tree.getModel()).removeNodeFromParent(clickedNode);
+                    for (int i = 0; i < clickedNode.getChildCount(); i++) {
```

Comment:
This whole code is blocking the UI. You need to run it in a background task

Reply:
Did You read Aaron's spec? Because it says: **NOTE:  Deleting documents that have blob attachments on them will require you to run the database maintenance API with the MaintenanceType of COMPACT.**

Once you finish deleting everything, you need to run this maintenance.  ActiveCBLDatabase.getInstance().getDatabase().performMaintenance(MaintenanceType.COMPACT);

Reply:
Performing maintenance after every scope/collection/document/blob deletion.